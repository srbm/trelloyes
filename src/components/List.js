import React from 'react';
import Card from './Card';
import './List.css';

function List(props) {
  const cardIds = props.cards.map((item, i) => 
    <Card key={i} title={item.title} content={item.content} />)
  return (
    <section className="List">
      <header className="List-header">
        <h2>{props.header}</h2>
      </header>
      <div className="List-cards">
        {cardIds}
      </div>
    </section>
  );
}

export default List;