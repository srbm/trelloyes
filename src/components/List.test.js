import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('List test suite', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List key={1} header="Test Header" cards={[1,2,3]} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected', () => {
    const tree = renderer
      .create(<List key={1} header="Test Header" cards={[1,2,3]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

