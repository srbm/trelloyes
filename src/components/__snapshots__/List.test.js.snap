// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List test suite renders UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Test Header
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
  </div>
</section>
`;
